### PR TITLE
Add basic support for built-in postgres point type

### PR DESCRIPTION
**Pull request by InAbsentia**
This is a rebased version of #126, updated to account for the upstream extension changes.

Thanks to @jbavari for doing the hard work of figuring out how to create the type and all the initial coding on this.

### CODE REVIEW

**Comment by jbavari** [comment]
@InAbsentia Thank you for getting it across the finish line!

Off topic: I see you're in Denver! We've got a new Elixir group going in Boulder if you want to come by. I'll be speaking next month! Come hang!

File: lib/postgrex/builtins.ex [hunk 1]
```diff
@@ -146,3 +146,19 @@ defmodule Postgrex.MACADDR do
...
+
+defmodule Postgrex.Point do
+  @moduledoc """
+  Struct for Postgres point.
+
+  ## Fields
+    * `x`
+    * `y`
+  """
+  require Decimal
```
Comment (ericmj) [inline lib/postgrex/builtins.ex]: :scissors: this line.

File: lib/postgrex/extensions/point.ex [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  @moduledoc false
+  import Postgrex.BinaryUtils
+  use Postgrex.BinaryExtension, send: "point_send"
+
+  def encode(_, %Postgrex.Point{x: x, y: y}, _, _),
+    do: <<x::float64, y::float64>>
+
+  def encode(type_info, value, _, _) do
+    raise ArgumentError,
+    Postgrex.Utils.encode_msg(type_info, value, Postgrex.Point)
```
Comment (ericmj) [inline lib/postgrex/extensions/point.ex]: Make the raise a single line or indent the second line.

File: lib/postgrex/utils.ex [hunk 3]
```diff
@@ -24,7 +24,8 @@ defmodule Postgrex.Utils do
     Postgrex.Extensions.Timestamp,
     Postgrex.Extensions.Interval,
     Postgrex.Extensions.UUID,
-    Postgrex.Extensions.Void]
+    Postgrex.Extensions.Void,
+    Postgrex.Extensions.Point]
```
Comment (ericmj) [inline lib/postgrex/utils.ex]: Please keep this list sorted by name.

File: test/query_test.exs [hunk 4]
```diff
@@ -109,6 +109,18 @@ defmodule QueryTest do
            query("SELECT interval '1 year 2 months 40 days 3 hours 2 minutes'", [])
   end
 
+  test "point comparison", context do
+    assert [[%Postgrex.Point{x: -97, y: 100}]] == [[%Postgrex.Point{x: -97.0, y: 100.0}]]
+  end
```
Comment (ericmj) [inline test/query_test.exs]: This test is not necessary.

**Comment by ericmj** [comment]
Looks good, I just dropped a few comments. Are you planning to do the other geometric types too?

**Comment by InAbsentia** [comment]
Thanks @ericmj. I'll make the changes and push an update. I'm not planning to add the other geometric types right now, but @jbavari and I are going to discuss working together on them in the near future.

**Comment by InAbsentia** [comment]
@ericmj Updated according to your comments.

**Comment by ericmj** [comment]
Beautiful :heart:!

**Comment by InAbsentia** [comment]
Thanks!

**Pull request merged**